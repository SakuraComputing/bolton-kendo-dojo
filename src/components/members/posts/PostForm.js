import React, { Component } from 'react';
import { connect } from 'react-redux';
import TextAreaFieldGroup from '../../../common/TextAreaFieldGroup';
import { addPost } from '../../../actions/postActions';

export class PostForm extends Component {

    constructor(props) {
        super(props)
        this.state = {
            text: '',
            errors: {}
        }
    }

    onChange = (e) => {
        this.setState({ text: e.target.value })
    }
    
    render() {

        const { errors } = this.state;

        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <div className="form-group">
                                <TextAreaFieldGroup
                                    className="post-text"
                                    placeholder="Create a post"
                                    name="text"
                                    value={this.state.text}
                                    onChange={this.onChange}
                                    error={errors.text}
                                />
                            </div>
                    <button type="submit" className="post-submit">
                        Submit
                    </button>                
                </form>
            </div>
        )
    }
}
const mapStateToProps = state => ({
    auth: state.auth,
    errors: state.errors
})
export default connect(mapStateToProps, { addPost })(PostForm);
