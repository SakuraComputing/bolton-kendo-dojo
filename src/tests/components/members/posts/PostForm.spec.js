import React from 'react';
import { shallow } from 'enzyme';
import { PostForm } from '../../../../components/members/posts/PostForm.js';

describe('Post Form Component', () => {
    
    let postForm;

    beforeEach(() => {
        postForm = shallow(<PostForm />)
    })
    it('should render correctly', () => {
        expect(postForm).toMatchSnapshot();
    });

    it('should set the initial state', () => {
        expect(postForm.state()).toEqual({ text: '', errors: {} })
    });

    it('should have a submit button', () => {
        expect(postForm.find('.post-submit').exists()).toBe(true);
    });

    it('should change the post text area when the user types', () => {
        const postText = 'Something';
        postForm.find('.post-text').simulate('change', {
            target: { value: postText }
        });
        expect(postForm.state().text).toEqual(postText);
    });
});