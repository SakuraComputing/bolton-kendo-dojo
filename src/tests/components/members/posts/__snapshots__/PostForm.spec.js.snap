// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Form Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <TextAreaFieldGroup
            name="text"
            onChange={[Function]}
            placeholder="Create a post"
            value=""
          />
          <button
            className="post-submit button-small"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-group"
        >
          <TextAreaFieldGroup
            name="text"
            onChange={[Function]}
            placeholder="Create a post"
            value=""
          />
          <button
            className="post-submit button-small"
            type="submit"
          >
            Submit
          </button>
        </div>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextAreaFieldGroup
              name="text"
              onChange={[Function]}
              placeholder="Create a post"
              value=""
            />,
            <button
              className="post-submit button-small"
              type="submit"
            >
              Submit
            </button>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "error": undefined,
              "name": "text",
              "onChange": [Function],
              "placeholder": "Create a post",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "post-submit button-small",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <TextAreaFieldGroup
              name="text"
              onChange={[Function]}
              placeholder="Create a post"
              value=""
            />
            <button
              className="post-submit button-small"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="form-group"
          >
            <TextAreaFieldGroup
              name="text"
              onChange={[Function]}
              placeholder="Create a post"
              value=""
            />
            <button
              className="post-submit button-small"
              type="submit"
            >
              Submit
            </button>
          </div>,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextAreaFieldGroup
                name="text"
                onChange={[Function]}
                placeholder="Create a post"
                value=""
              />,
              <button
                className="post-submit button-small"
                type="submit"
              >
                Submit
              </button>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "error": undefined,
                "name": "text",
                "onChange": [Function],
                "placeholder": "Create a post",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "post-submit button-small",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
